packages/cli/src/UserManagement/routes/users.ts

```diff
@@ -203,4 +201,155 @@ export function addUsersMethods(this: N8nApp): void {
+				await Db.collections.SharedWorkflow!.update({ user: deletedUser }, { user: transferUser });
+				await Db.collections.SharedCredentials!.update(
+					{ user: deletedUser },
+					{ user: transferUser },
+				);
+			} else {
+				const queryBuilderWorkflows = Db.collections.Workflow!.createQueryBuilder('w');
+				queryBuilderWorkflows.select(['w.id']);
+				queryBuilderWorkflows.innerJoin('w.shared', 'shared');
+				queryBuilderWorkflows.andWhere('shared.userId = :userId', {
+					userId: deletedUser.id,
+				});
+
+				const workflows = await queryBuilderWorkflows.getMany();
+
+				if (workflows.length) {
+					await Db.collections.Workflow!.remove(workflows);
+					await Db.collections.SharedWorkflow!.delete({ user: deletedUser });
+				}
+
+				const queryBuilderCredentials = Db.collections.Credentials!.createQueryBuilder('c');
+				queryBuilderCredentials.innerJoin('c.shared', 'shared');
+				queryBuilderCredentials.andWhere('shared.userId = :userId', {
+					userId: deletedUser.id,
+				});
+
+				queryBuilderCredentials.select(['c.id']);
+
+				const credentials =
+					(await queryBuilderCredentials.getMany()) as unknown as ICredentialsResponse[];
+
+				if (credentials.length) {
+					await Db.collections.Credentials!.remove(credentials);
+					await Db.collections.SharedCredentials!.delete({ user: deletedUser });
+				}
+			}
+			await Db.collections.User!.delete({ id: deletedUser.id });
+			return { success: true };
+		}),
+	);
+
+	this.app.post(
+		`/${this.restEndpoint}/users/:id/reinvite`,
+		ResponseHelper.send(async (req: Request) => {
+			if ((req.user as User).globalRole.name !== 'owner') {
+				throw new ResponseHelper.ResponseError(
+					'Current user cannot perform this operation',
+					undefined,
+					403,
+				);
+			}
+
+			if (!isEmailSetup()) {
+				throw new ResponseHelper.ResponseError(
+					'Email sending must be set up in order to invite other users',
+					undefined,
+					500,
+				);
+			}
+
+			const user = await Db.collections.User!.findOne({ id: req.params.id });
+
+			if (!user) {
+				throw new ResponseHelper.ResponseError('User not found', undefined, 403);
+			}
+
+			if (user.password) {
+				throw new ResponseHelper.ResponseError(
+					'User has already accepted the invite',
+					undefined,
+					403,
+				);
+			}
+
+			let inviteAcceptUrl = GenericHelpers.getBaseUrl();
+			const domain = inviteAcceptUrl;
+			if (!inviteAcceptUrl.endsWith('/')) {
+				inviteAcceptUrl += '/';
+			}
+			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-unsafe-member-access
+			inviteAcceptUrl += `signup/inviterId=${(req.user as User).id}&inviteeId${user.id}`;
+
+			const mailer = getInstance();
+			// eslint-disable-next-line no-await-in-loop
+			const result = await mailer.invite({
+				email: user.email,
+				inviteAcceptUrl,
+				domain,
+			});
+
+			if (result.success) {
+				return { success: true };
+			}
+			return { success: false };
```

Comment:
shouldn't we return a matching status code then? because I think the FE will not use the data here